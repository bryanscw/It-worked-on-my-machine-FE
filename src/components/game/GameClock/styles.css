.clock {
    position: absolute;
    top: 5px;
    right: 5px;
    background-color: #ffffffdd;
    padding: 5px;
    border-radius: 5px;
}