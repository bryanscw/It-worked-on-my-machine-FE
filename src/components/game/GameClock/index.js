import React from "react";
import formatDuration from "format-duration";

import PokemonClock from "./pokemonClock.png";
import "./styles.css";

export default function GameClock(props) {
    const {
        elapsedTime,
        penaltyTime,
    } = props;

    return (
        <div className="clock d-flex align-items-center">
            <img
                src={PokemonClock}
                alt="Pokemon Clock"
                width="20px"
                height="20px"
                className="mr-2"
            />
            <small className="mr-2">
                {formatDuration(elapsedTime)}
            </small>
            <small className="text-danger">
                Penalty: {penaltyTime / 1000}s
            </small>
        </div>
    );
}
