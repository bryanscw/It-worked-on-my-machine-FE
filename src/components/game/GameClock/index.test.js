import React from "react";
import GameClock from "./index.js";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import ReactDOM from "react-dom";

afterEach(cleanup);

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<GameClock elapsedTime = {0} penaltyTime = {0} />, div);
});

it('should take a snapshot', async () => {
    const elapsedTime = 0;
    const penaltyTime = 0;
    const { asFragment, getByText } = render(<GameClock elapsedTime = {0} penaltyTime = {0} />);
    expect(asFragment()).toMatchSnapshot();
})