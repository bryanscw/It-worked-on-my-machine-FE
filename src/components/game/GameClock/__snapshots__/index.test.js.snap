// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="clock d-flex align-items-center"
  >
    <img
      alt="Pokemon Clock"
      class="mr-2"
      height="20px"
      src="pokemonClock.png"
      width="20px"
    />
    <small
      class="mr-2"
    >
      0:00
    </small>
    <small
      class="text-danger"
    >
      Penalty: 0s
    </small>
  </div>
</DocumentFragment>
`;
